sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClient.java

```diff
@@ -581,49 +593,56 @@ public GlobalThroughputControlConfigBuilder createGlobalThroughputControlConfigB
-    }
-
     WriteRetryPolicy getNonIdempotentWriteRetryPolicy() {
         return this.nonIdempotentWriteRetryPolicy;
     }
 
-    private Mono<List<Void>> openConnectionsAndInitCachesInternal() {
-        int concurrency = 1;
-        int prefetch = 1;
-        if (this.proactiveContainerInitConfig != null) {
-            return Flux.fromIterable(this.proactiveContainerInitConfig.getCosmosContainerIdentities())
-                    .flatMap(
-                        cosmosContainerIdentity -> Mono.just(this
-                            .getDatabase(containerIdentityAccessor.getDatabaseName(cosmosContainerIdentity))
-                            .getContainer(containerIdentityAccessor.getContainerName(cosmosContainerIdentity))),
-                        concurrency,
-                        prefetch
-                    )
-                    .flatMap(
-                        cosmosAsyncContainer -> cosmosAsyncContainer
-                            .openConnectionsAndInitCaches(
-                                this.proactiveContainerInitConfig.getProactiveConnectionRegionsCount()),
-                        concurrency,
-                        prefetch)
-                    .collectList();
-        }
-        return Mono.empty();
+    void openConnectionsAndInitCaches() {
+        final Duration lastSuccessResponseTimeout = Duration.ofSeconds(5);
+
+        asyncDocumentClient.submitOpenConnectionTasksAndInitCaches(
+                        proactiveContainerInitConfig,
+                        AsyncDocumentClient.OpenConnectionAggressivenessHint.AGGRESSIVE
+                )
+                .subscribeOn(CosmosSchedulers.OPEN_CONNECTIONS_BOUNDED_ELASTIC)
+                .subscribe();
+
+        wrapSourceFluxWithSoftTimeoutAndFallback(
+                proactiveOpenConnectionsProcessor.getOpenConnectionsPublisher().sequential(),
+                Flux.empty(),
+                lastSuccessResponseTimeout,
+                CosmosSchedulers.OPEN_CONNECTIONS_BOUNDED_ELASTIC
+        ).blockLast();
     }
 
-    private void blockListVoidResponse(Mono<List<Void>> voidListMono) {
-        try {
-            voidListMono.block();
-        } catch (Exception ex) {
-            final Throwable throwable = Exceptions.unwrap(ex);
 
-            if (throwable instanceof CosmosException) {
-                throw (CosmosException) throwable;
-            } else {
-                throw Exceptions.propagate(throwable);
-            }
-        }
+    void openConnectionsAndInitCaches(Duration timeout) {
+        asyncDocumentClient.submitOpenConnectionTasksAndInitCaches(
+                        proactiveContainerInitConfig,
+                        AsyncDocumentClient.OpenConnectionAggressivenessHint.AGGRESSIVE
+                )
+                .subscribeOn(CosmosSchedulers.OPEN_CONNECTIONS_BOUNDED_ELASTIC)
+                .subscribe();
+
+        proactiveOpenConnectionsProcessor
+                .getOpenConnectionsPublisher()
```

Comment:
do we need to subscribeOn the open connection scheduler?

Reply:
No harm in adding `subscribeOn` since it applies to all upstream operators.